src/natcap/invest/utils.py

```diff
@@ -538,6 +541,38 @@ def build_lookup_from_csv(
     return lookup_dict 
 
 
+def read_csv_to_dataframe(path, to_lower=False, encoding=None, **kwargs):
+    """
+    Wrapper around ``pandas.read_csv`` that standardizes the column names by
+    stripping leading/trailing whitespace and optionally making all lowercase.
+    This helps avoid common errors caused by user-supplied CSV files with
+    column names that don't exactly match the specification.
+
+    Args:
+        path (string): path to a CSV file
+        to_lower (bool): if True, convert all column names to lowercase
+        encoding (string): name of encoding codec to pass to `read_csv`. 
+            Defaults to None.
+        **kwargs: any kwargs that are valid for ``pandas.read_csv``
+    """
+    # Check if the file encoding is UTF-8 BOM first
+    # allow encoding kwarg to override this if it's provided
+    if not encoding:
+        with open(path, 'rb') as file_obj:
+            first_line = file_obj.readline()
+            if first_line.startswith(codecs.BOM_UTF8):
+                encoding = 'utf-8-sig'
+    dataframe = pandas.read_csv(path, encoding=encoding, **kwargs)
+    # this won't work on non-string types
+    try:
+        dataframe.columns = dataframe.columns.str.strip()
+        if to_lower:
+            dataframe.columns = dataframe.columns.str.lower()
+    except:
+        pass
```

Comment:
Could I ask about this?  Is there an error that we're expecting to handle here?  Or is this a 'just in case' sort of exception handler?  When could we expect this handler to be triggered?

The reason I'm asking is that if there's a specific exception that we're expecting will be raised, it would be better to catch that specific exception and handle it appropriately.  If this is the case (and depending on the error), it might also be worth describing the circumstances under which the exception might be raised.

But the other big thing is that errors should not pass silently ... that'll come back to bit us _hard_ in the future when we can't find out exactly what happened from the log or immediately from the input files.  I think it'd be more maintainable for us in the long run for this to catch the specific exception we expect, or if this is really a clause that's likely to be triggered, I'd suggest that we add a `LOGGER.exception` here.

Reply:
And if this is related to the note above the block about not working on string types, can we replicate this in the tests?

Reply:
I originally did this for the case where `header=None`, which cause the `dataframe.columns` to be an Index with dtype int64, so the string operations didn't work. Later decided there's no point in using this method at all if `header=None`, but kept the `try` block to handle the case where the CSV does have non-strings for some reason. I will make the exception more specific though.